accounts-db/src/ring/ring_dir_remover.rs

```diff
@@ -0,0 +1,262 @@
+use {
+    crate::ring::{Ring, RingCtx, RingOp},
+    io_uring::{opcode, squeue, types, IoUring},
+    slab::Slab,
+    std::{
+        collections::VecDeque,
+        ffi::{CStr, OsString},
+        fs, io,
+        os::{
+            fd::{AsRawFd, IntoRawFd, OwnedFd, RawFd},
+            unix::ffi::OsStringExt as _,
+        },
+        path::PathBuf,
+    },
+};
+
+/// A fast directory remover using io_uring.
+pub struct RingDirRemover {
+    ring: Ring<State, Op>,
+}
+
+impl RingDirRemover {
+    /// Creates a new remover.
+    pub fn new() -> io::Result<RingDirRemover> {
```

Comment:
this will fail if kernel doesn't support io_uring?

Reply:
yes, build() will return io::Error doing the IO_URING_SETUP syscall

Reply:
I've added an explicit `io_uring_supported` so we can test for the exact set of features we require